src/rp2_common/pico_status_led/include/pico/status_led.h

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2025 Raspberry Pi (Trading) Ltd.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/** \file pico/status_led.h
+ *  \defgroup pico_status_led pico_status_led
+ *
+ * \brief Enables access to the on-board status leds
+ *
+ * Boards usually have access to an on-board status leds which are configured via the board header (\see PICO_DEFAULT_LED_PIN and \see PICO_DEFAULT_WS2812_PIN)
+ * This library hides the details so you can use the status leds for all boards without changing your code.
+ */
+
+#ifndef _PICO_STATUS_LED_H
+#define _PICO_STATUS_LED_H
+
+#include "hardware/gpio.h"
+
+struct async_context;
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// PICO_CONFIG: PICO_STATUS_LED_WS2812_WRGB, Inidicate if the colored status led supports WRGB, type=bool, default=false group=pico_status_led
+#ifndef PICO_STATUS_LED_WS2812_WRGB
+#define PICO_STATUS_LED_WS2812_WRGB 0
+#endif
+
+/*! \brief Initialise the status leds
+ *  \ingroup pico_status_led
+ *
+ *  Initialise the status leds and the resources they need before use.
+ *
+ * \param context The hardware for some devices (e.g. Pico W) requires an async context.
+ * You can usually only have one of these. Pass the async context into the function or pass NULL to get the function to just create a context for it's own use if needed .
+ * \return Returns true if the led was initialised successfully, otherwise an error occurred
+ */
+bool pico_status_led_init(struct async_context *context);
+
+/*! \brief Set the status led on or off
+ *  \ingroup pico_status_led
+ *
+ * \note: If your hardware does not support a status led (\see PICO_DEFAULT_LED_PIN), this function does nothing and returns false.
+ *
+ * \param led_on True to turn the led on. Pass False to turn the led off
+ * \param True if the status led could be set, otherwise false
+ */
+static inline bool pico_status_led_set(bool led_on) {
+#if defined PICO_DEFAULT_LED_PIN
+    gpio_put(PICO_DEFAULT_LED_PIN, led_on);
+#elif defined CYW43_WL_GPIO_LED_PIN
+    cyw43_arch_gpio_put(CYW43_WL_GPIO_LED_PIN, led_on);
```

Comment:
Not sure I understand how this links as it's in pico_cyw43_arch. we need to include the right header file in here?